newIDE/app/src/UI/Carousel.js

```diff
@@ -0,0 +1,375 @@
+        currentFirstVisibleItemIndex >
+          itemsToDisplay.length - visibleThumbnailsCount - 1
+      )
+        return 0;
+      return roundScroll(
+        scrollViewElement.scrollLeft + scale * (direction === 'left' ? -1 : 1)
+      );
+    },
+    [widthUnit, itemsToDisplay]
+  );
+
+  const roundScroll = React.useCallback(
+    (value: number): number => {
+      return Math.round(value / widthUnit) * widthUnit;
+    },
+    [widthUnit]
+  );
+
+  const onClickArrow = (direction: 'left' | 'right') =>
+    React.useCallback(
+      (): void => {
+        const scrollViewElement = scrollView.current;
+        if (!scrollViewElement) return;
+        const newScrollPosition = computeScroll(direction, scrollViewElement);
+
+        scrollViewElement.scrollTo({
+          left: newScrollPosition,
+          behavior: 'smooth',
+        });
+      },
+      [computeScroll]
+    );
+
+  const handleScroll = React.useCallback(() => {
+    const scrollViewElement = scrollView.current;
+    if (!scrollViewElement) return;
+
+    if (!shouldDisplayLeftArrow)
+      setShouldDisplayLeftArrow(scrollViewElement.scrollLeft !== 0);
+  });
+  const handleScrollEnd = React.useCallback(
+    () => {
+      const scrollViewElement = scrollView.current;
+      if (!scrollViewElement) return;
+
+      scrollViewElement.scrollTo({
+        left: roundScroll(scrollViewElement.scrollLeft),
+        behavior: 'smooth',
+      });
+    },
+    [roundScroll]
+  );
+
+  const onFocusItem = event => {
+    event.target.scrollIntoViewIfNeeded();
+  };
+
+  React.useEffect(() => {
+    const scrollViewElement = scrollView.current;
+    if (!scrollViewElement) return;
+
+    // Add event listeners on component mount. There is no need to
+    // remove them with a cleanup function because scrollview element
+    // does not change and they will be destroyed when the element is
+    // removed from the DOM.
+    scrollViewElement.addEventListener('scroll', handleScroll);
+    scrollViewElement.addEventListener('touchend', handleScrollEnd);
+    scrollViewElement.addEventListener('touchleave', handleScrollEnd);
+  }, []);
+
+  return (
+    <Line noMargin expand>
+      <div
+        style={{ ...styles.arrowContainer, width: arrowWidth }}
+        onClick={onClickArrow('left')}
+      >
+        {areItemsSet && shouldDisplayLeftArrow && <ArrowBackIos />}
+      </div>
+      <div
+        style={{
+          width: `calc(100% - ${2 * arrowWidth}px - ${rightArrowMargin}px)`,
+        }}
+      >
+        <Line noMargin justifyContent="space-between" alignItems="center">
+          <Text size="bold-title">{title}</Text>
+          <FlatButton
+            onClick={
+              onBrowseAllClick ||
+              (browseAllLink
+                ? () => {
+                    Window.openExternalURL(browseAllLink);
+                  }
+                : () => {})
+            }
+            label={<Trans>Browse all</Trans>}
+            tabIndex={0}
```

Comment:
Remove the tabIndex for this button, it's a button so it should by default properly keyboard focusable